generate-clang-meson-cross-file.sh

```diff
@@ -70,7 +72,9 @@ strip = '${toolchain_dest}/bin/${toolchain_target}-strip'
 needs_exe_wrapper = true
 has_function_printf = false
 c_args = [${meson_cflags}]
+c_link_args = [${meson_cflags}]
 cpp_args = [${meson_cflags}]
+cpp_link_args = [${meson_cflags}]
```

Comment:
This concerns me a bit. It's true that some compilation options also need to be passed as link options, but others don't make sense. Is there a reason for why `meson_cflags` will always contain options that make sense for both compilation and linking? If not, can't we further separate this?
BTW, can you specify which options were missing? That might shed some light on the best way forward.

Reply:
I've been very careful to ensure only toolchain-related flags end up in `meson_cflags`.

What is a toolchain-related flag? One that is needed to ensure the compiler always compiles in a way that is compatible with the toolchain, or to ensure it uses the toolchain. This list currently is
- `-march`
- `-mabi`
- `-mcmodel`
- `--gcc-toolchain` (so clang does lookup correctly)

Reply:
I was already assuming that you had done a good job with such separation :-) My point was that I was concerned that, given the name `meson_cflags`, it wouldn't be clear in the future. This was coming from the perspective of CFLAGS and LDFLAGS naming, where the CFLAGS content isn't necessarily applicable to LDFLAGS, so I was concerned that it wouldn't be clear that you couldn't add stuff to `meson_cflags` that would also apply to the linking stage. If you think this isn't going to be a problem or that there isn't an easy way to address this potential pitfall, then that's good enough :-)

Reply:
Ah, I see where you're going with this. Maybe we should call it `meson_driver_flags`?

Reply:
Sounds good!